okx: fix type for AccountConfigurationResponse

junnplus opened the pull request:
# PR Description

This pull request includes changes to the `AccountConfigurationResponse` struct in the `okx_types.go` file to correct data types for specific fields. The most important changes are:

* Changed the type of `Level` from `types.Number` to `string` to correctly represent the user level as a string.
* Changed the type of `TraderInsts` from `string` to `[]string` to correctly represent multiple trader instructions as a list of strings.
* Changed the type of `SpotTraderInsts` from `string` to `[]string` to correctly represent multiple spot trader instructions as a list of strings.

https://www.okx.com/docs-v5/en/#trading-account-rest-api-get-account-configuration

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

shazbert commented on exchanges/okx/okx_types.go:
```diff
@@ -1922,31 +1922,31 @@ type BillsDetailResponse struct {
+	DiscountType                   string   `json:"discountType"`
+	LiquidationGear                string   `json:"liquidationGear"`
+	EnableSpotBorrow               bool     `json:"enableSpotBorrow"`
+	SpotBorrowAutoRepay            bool     `json:"spotBorrowAutoRepay"`
```
Can you please add in field Type types.Number `json:"type"`

<img width="470" alt="image" src="https://github.com/user-attachments/assets/87cb0c69-7948-48c9-8a52-e48403eba24b" />

shazbert reviewed (changes requested):
@junnplus Thank you so much for this PR! Just some things I have found while you are here. 🚀

junnplus commented:
@shazbert Thanks for the review! This PR mainly addresses issues where incorrect types prevent unmarshaling. For string-to-number conversions, I prefer handling them case by case based on specific scenarios.

shazbert reviewed (approved):
> @shazbert Thanks for the review! This PR mainly addresses issues where incorrect types prevent unmarshaling. For string-to-number conversions, I prefer handling them case by case based on specific scenarios.

Deferring OOS requests. Changes ACK. Have to verify so couldn't test.

gloriousCode reviewed (approved):
tACK! Thank you for letting VIPs retrieve account information 

### 🎩 
### 🧐

thrasher- commented:
> @shazbert Thanks for the review! This PR mainly addresses issues where incorrect types prevent unmarshaling. For string-to-number conversions, I prefer handling them case by case based on specific scenarios.

Looking at all the fields, in all cases they return an integer so we won't have any issues with a loss of precision in this context. From testing, there's two fields which can be removed (DiscountType and SpotOffsetType) whilst here. I'm happy to push these changes or provide a diff. That way we can resolve all issues in one pass.

thrasher- commented:
@junnplus patch supplied here: https://gist.githubusercontent.com/thrasher-/1174e899ac635b82b0ae5cf573419e44/raw/bc95de73704d076e57ddd45e37aabc53e53ee0fc/okx_account_config.patch

Or `curl https://gist.githubusercontent.com/thrasher-/1174e899ac635b82b0ae5cf573419e44/raw/bc95de73704d076e57ddd45e37aabc53e53ee0fc/okx_account_config.patch | git apply`

junnplus commented:
@thrasher- Thank you for the patch! You can directly push your changes to this PR. This way, your commit signature will be preserved.

shazbert reviewed (approved):
YAY! 🚀 Thanks.

thrasher- commented:
> @thrasher- Thank you for the patch! You can directly push your changes to this PR. This way, your commit signature will be preserved.

Thank you sir 🙏

gbjk reviewed (approved):
LGTM 👍
